Others/Dijkstra.java

```diff
@@ -0,0 +1,178 @@
+package Others;
+
+/**
+ * Dijkstra's algorithm,is a graph search algorithm that solves the single-source
+ * shortest path problem for a graph with nonnegative edge path costs, producing
+ * a shortest path tree.
+ * 
+ * NOTE:  The inputs to Dijkstra's algorithm are a directed and weighted graph consisting
+ * of 2 or more nodes, generally represented by an adjacency matrix or list, and a start node.
+ * 
+ * Original source of code: https://rosettacode.org/wiki/Dijkstra%27s_algorithm#Java
+ * Also most of the comments are from RosettaCode.
+ *
+ */
+
+//import java.io.*;
+import java.util.*; 
+public class Dijkstra {
+   private static final Graph.Edge[] GRAPH = {
+      new Graph.Edge("a", "b", 7),	//Distance from node "a" to node "b" is 7. In the current Graph there is no way to move the other way (e,g, from "b" to "a"), a new edge would be needed for that
+      new Graph.Edge("a", "c", 9),
+      new Graph.Edge("a", "f", 14),
+      new Graph.Edge("b", "c", 10),
+      new Graph.Edge("b", "d", 15),
+      new Graph.Edge("c", "d", 11),
+      new Graph.Edge("c", "f", 2),
+      new Graph.Edge("d", "e", 6),
+      new Graph.Edge("e", "f", 9),
+   };
+   private static final String START = "a";
+   private static final String END = "e";
+
+	/**
+	 * main function
+	 * Will run the code with "GRAPH" that was defined above.
+	 */
+   public static void main(String[] args) {
+      Graph g = new Graph(GRAPH);
+      g.dijkstra(START);
+      g.printPath(END);
+      //g.printAllPaths();
+   }
+}
+ 
+class Graph {
+   private final Map<String, Vertex> graph; // mapping of vertex names to Vertex objects, built from a set of Edges
+ 
+   /** One edge of the graph (only used by Graph constructor) */
+   public static class Edge {
+      public final String v1, v2;
+      public final int dist;
+      public Edge(String v1, String v2, int dist) {
+         this.v1 = v1;
+         this.v2 = v2;
+         this.dist = dist;
+      }
+   }
+ 
+   /** One vertex of the graph, complete with mappings to neighbouring vertices */
+  public static class Vertex implements Comparable<Vertex>{
+	public final String name;
+	public int dist = Integer.MAX_VALUE; // MAX_VALUE assumed to be infinity
+	public Vertex previous = null;
+	public final Map<Vertex, Integer> neighbours = new HashMap<>();
+ 
+	public Vertex(String name)
+	{
+		this.name = name;
+	}
+ 
+	private void printPath()
+	{

```

Comment:
The opening brace ` {` and method declarations should be on the same line. Most of the standard source code for the Java API is written in this style. 
eg.
```java
public Vertex(String name) {
    this.name = name;
}
```

Reply:
Also for the blocks inside a function.
```java
if (this == this.previous) {
    System.out.printf("%s", this.name);
}